python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -629,6 +645,85 @@ def unpack(
         check.failed(f"Unexpected class {self.klass}")
 
 
+@whitelist_for_serdes
+class SkipOnNotAllParentsUpdatedRule(
+    AutoMaterializeRule,
+    NamedTuple(
+        "_SkipOnNotAllParentsUpdatedRule", [("require_update_on_all_parent_partitions", bool)]
+    ),
+):
+    """An auto-materialize rule that enforces that an asset can only be requested for execution
+    if all parents have been materialized since its last materialization.
+
+    Properties:
+        require_update_on_all_parent_partitions (Optional[bool]): If true, requires all upstream
+            partitions to be materialized in order for a given asset to update. If false, requires
+            at least one upstream partition in each parent asset to be materialized in order for
+            a given asset to update. Defaults to false.
+    """
+
```

Comment:
add description `@property` here

this is what will be rendered in the UI, I'm thinking something along the lines of "waiting for all parents to update" (see SkipOnParentMissingRule as an example)

Reply:
Updated!
- In the default "any partition" case, we will display "waiting on upstream data to be updated"
- In the "all" case we will display "waiting until all upstream partitions are updated"